[IsaacPangTH] iP

IsaacPangTH opened the pull request:
# Him

>"I'm him" -Him

- With GUI!
- Easy to use
- He's him

Helps you keep track of:
1. Todos
2. Deadlines
3. Events

Most importantly he does all this for **FREE**:boom::boom::boom:
Just download it from [here](https://github.com/IsaacPangTH/ip)

Features:
- [x] Main features
- [ ] Extensions

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

IsaacPangTH committed:
Renamed main class

IsaacPangTH committed:
added greeting and exit

IsaacPangTH committed:
Added support for echo

IsaacPangTH committed:
created HimList class

IsaacPangTH committed:
Added add and list functionality

IsaacPangTH committed:
Added Task class

IsaacPangTH committed:
Implemented HimList with Tasks instead of Strings

IsaacPangTH committed:
Added mark

IsaacPangTH committed:
added todo

IsaacPangTH committed:
fixed bug from todos with spaces

IsaacPangTH committed:
added dealine

IsaacPangTH committed:
added events

IsaacPangTH committed:
set up TextUiTesting

IsaacPangTH committed:
added catches for invalid arguments

IsaacPangTH committed:
created AlreadyCompletedException and TaskDoesNotExistException

IsaacPangTH committed:
handled unknoown commands and empty lists

IsaacPangTH committed:
updated testing to test for handled errors

IsaacPangTH committed:
added delete

IsaacPangTH committed:
Refactor code

IsaacPangTH committed:
Add storage package

IsaacPangTH committed:
Add save

IsaacPangTH committed:
Add test for save

IsaacPangTH committed:
Add save data to gitignore

IsaacPangTH committed:
Merge branch 'branch-Level-7'

IsaacPangTH committed:
Change Deadline to store due date as LocalDate instead of String

IsaacPangTH committed:
Change format of deadline toString

IsaacPangTH committed:
Add check for events which start after they end

IsaacPangTH committed:
Merge branch 'branch-Level-8'

IsaacPangTH committed:
Update text-ui-test

IsaacPangTH committed:
Merge branch 'branch-Level-8'

IsaacPangTH committed:
Add catch for invalid arguments in event creation

IsaacPangTH committed:
Merge branch 'branch-Level-8'

IsaacPangTH committed:
Add Ui class

IsaacPangTH committed:
Add new exceptions

IsaacPangTH committed:
Add commands

IsaacPangTH committed:
Fix bug in UI causing inconsistent new lines

IsaacPangTH committed:
Add parser

IsaacPangTH committed:
Refactor Storage into him package

IsaacPangTH committed:
Remove unused methods and fields

IsaacPangTH committed:
Merge remote-tracking branch 'origin/add-gradle-support'

IsaacPangTH committed:
Configure Gradle

IsaacPangTH committed:
Add JUnit tests for Parser

IsaacPangTH committed:
Fix bugs

IsaacPangTH committed:
Add JUnit tests for TaskList complete method

IsaacPangTH committed:
Configure Shadow Gradle plugin

IsaacPangTH committed:
Reformat code

IsaacPangTH committed:
Add Javadoc

IsaacPangTH committed:
Add find feature

IsaacPangTH committed:
Merge branch 'branch-A-CodingStandard'

IsaacPangTH committed:
Merge branch 'branch-Level-9'

1st2GetThisName commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,29 @@
+    private Task task;
+
+    /**
+     * Constructor for<code>AddCommand</code>.
```
Why is there no space between "for" and the HTML tag?

1st2GetThisName commented on src/main/java/him/Parser.java:
```diff
@@ -0,0 +1,138 @@
+import task.Event;
+import task.ToDo;
+
+import java.time.format.DateTimeParseException;
```
I appreciate how neatly you've arranged your import statements, really does demonstrate why they asked us to organize them like this.

1st2GetThisName reviewed (commented):
All in all, LGTM. Wasn't really able to find anything negative to say about the formatting, good job!

IsaacPangTH committed:
Fix bugs

Nimastic commented on src/main/java/command/ListCommand.java:
LGTM

Nimastic reviewed (commented):
LGTM

IsaacPangTH committed:
Set up Java FX

IsaacPangTH committed:
Add GUI

IsaacPangTH committed:
Reimplement greeting and exit command

IsaacPangTH committed:
Clean up code

jowhee3011 commented on src/main/java/command/ExitCommand.java:
```diff
@@ -0,0 +1,12 @@
+import him.Ui;
+import task.TaskList;
+
+public class ExitCommand extends Command {
```
Javadocs can be added here

jowhee3011 commented on src/main/java/exceptions/AlreadyCompletedException.java:
```diff
@@ -0,0 +1,16 @@
+    /**
+     * Constructor for<code>AlreadyCompletedException</code>
+     */
+    public AlreadyCompletedException() {
```
I like that this case is considered.

jowhee3011 commented on src/main/java/exceptions/StartAfterEndException.java:
```diff
@@ -0,0 +1,16 @@
+ *
+ * @author IsaacPangTH
+ */
+public class StartAfterEndException extends HimException {
```
Another case considered that I liked! Good job.

jowhee3011 reviewed (commented):
Generally LGTM! There are certain areas where javaodocs aren't there or clear enough. Otherwise, the code is very well structured. Good job!

IsaacPangTH committed:
Add assertions

IsaacPangTH committed:
Merge pull request #2 from IsaacPangTH/branch-A-Assertions

Add assertions

IsaacPangTH committed:
Refactor code

There are instances where code quality is suboptimal.

Code of suboptimal quality will make future maintenance difficult.

Extract methods, give appropriate names to variables
and add whitespace between logical blocks to improve readability.

Refactoring the code in these ways improves code quality.

IsaacPangTH committed:
Merge pull request #4 from IsaacPangTH/branch-A-CodeQuality

Refactor code

IsaacPangTH committed:
Add archive feature

There is no way to keep old tasks but not have them in the task list.

Users may want to keep old tasks but not see it on the main task list.

Add archive ArrayList field within TaskList class which can be used to
place archive tasks in.

Having the archive be accessible through the TaskList allows commands
to have access to the archive.

IsaacPangTH committed:
Merge pull request #5 from IsaacPangTH/branch-C-Archive

Add archive feature

IsaacPangTH committed:
Add title

IsaacPangTH committed:
Remove dead code

IsaacPangTH committed:
Add Ui.png

IsaacPangTH committed:
Add User Guide

IsaacPangTH committed:
Remove template text from UG

IsaacPangTH committed:
Fix line breaks in UG

IsaacPangTH committed:
Add Feature List to UG

IsaacPangTH committed:
Replace alert with heading in UG

IsaacPangTH committed:
Add missing colon in UG

IsaacPangTH committed:
Add handling for corrupted data files

IsaacPangTH committed:
Add quick start and contents to UG

IsaacPangTH committed:
Fix spacing inconsistency in UG

IsaacPangTH committed:
Improve GUI

The GUI is basic and not aesthetically pleasing.

A better GUI is needed for a better user experience.

Add a background color and speech bubbles through a css file.
Anchor UI elements and implement automatic resizing of UI elements
to allow proper window resizing.

Using CSS allows the colors and other visual properties
to be changed easily in the future.

IsaacPangTH committed:
Update Say method in Ui class

IsaacPangTH committed:
Update Ui.png

IsaacPangTH committed:
Update Ui.png
